### PR TITLE
READY: Show help message for docker image generation for integration tests

### PR DESCRIPTION
**Pull request by greenhat**
Show help message on "missing docker image" exception in integration tests when the first run is made not via sbt.

### CODE REVIEW

**Comment by greenhat** [comment]
Found some annoying case which will hit newcomers if they don't start with `sbt`. @mike-aksarin does it make any sense?

**Comment by greenhat** [comment]
`ErgoMinerSpec` failed with:
```
[0m[[0m[0minfo[0m] [0m[0m[32mErgoMinerSpec:[0m[0m
[pool-1-thread-1-ScalaTest-running-ErgoMinerSpec] >> [WARN ] o.e.settings.ErgoSettings$ >> 12:18:26.118 NO CONFIGURATION FILE WAS PROVIDED. STARTING WITH DEFAULT SETTINGS FOR TESTNET!
[default-akka.actor.default-dispatcher-7] >> [INFO ] o.e.nodeView.state.ErgoState$ >> 12:18:26.131 Generating genesis UTXO state
[default-akka.actor.default-dispatcher-7] >> [INFO ] s.c.a.a.b.VersionedIODBAVLStorage >> 12:18:26.302 Update storage to version ByteArrayWrapper[F2343E160D4E42A83A87EA1A2F56B6FA2046AB8146C5E61727C297BE578DA0F510]: 20006 elements to insert, 1 elements to remove
[default-akka.actor.default-dispatcher-7] >> [INFO ] o.e.nodeView.state.ErgoState$ >> 12:18:27.140 Genesis UTXO state generated
[scala-execution-context-global-14] >> [WARN ] s.core.utils.NetworkTimeProvider >> 12:18:44.696 Problems with NTP: 
java.net.SocketTimeoutException: Receive timed out
	at java.base/java.net.PlainDatagramSocketImpl.receive0(Native Method)
	at java.base/java.net.AbstractPlainDatagramSocketImpl.receive(AbstractPlainDatagramSocketImpl.java:180)
	at java.base/java.net.DatagramSocket.receive(DatagramSocket.java:814)
	at org.apache.commons.net.ntp.NTPUDPClient.getTime(NTPUDPClient.java:91)
	at org.apache.commons.net.ntp.NTPUDPClient.getTime(NTPUDPClient.java:113)
	at scorex.core.utils.NetworkTimeProvider.updateOffSet(NetworkTime.scala:33)
	at scorex.core.utils.NetworkTimeProvider.$anonfun$timeAndState$1(NetworkTime.scala:50)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:653)
	at scala.util.Success.$anonfun$map$1(Try.scala:251)
	at scala.util.Success.map(Try.scala:209)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:287)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:29)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:29)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:140)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:283)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1603)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175)
[scala-execution-context-global-14] >> [INFO ] s.core.utils.NetworkTimeProvider >> 12:18:44.697 New offset adjusted: None
[scala-execution-context-global-14] >> [INFO ] s.core.utils.NetworkTimeProvider >> 12:18:44.699 New offset adjusted: Some(0)
[scala-execution-context-global-14] >> [INFO ] s.core.utils.NetworkTimeProvider >> 12:18:44.700 New offset adjusted: Some(0)
[scala-execution-context-global-14] >> [INFO ] s.core.utils.NetworkTimeProvider >> 12:18:44.701 New offset adjusted: Some(0)
[default-akka.actor.default-dispatcher-3] >> [INFO ] org.ergoplatform.local.ErgoMiner >> 12:18:45.732 Starting Mining
[0m[[0m[0minfo[0m] [0m[0m[31m- should not freeze while generating candidate block with large amount of txs *** FAILED ***[0m[0m
[0m[[0m[0minfo[0m] [0m[0m[31m  An unexpected java.util.NoSuchElementException was thrown. (ErgoMinerSpec.scala:63)[0m[0m
```
@mike-aksarin Seems like a flaky test. I ran travis job locally and it did just fine. Can I trigger travis job via magic phrase from comments?
Also, travis does not like our long logs and refuses to parse it. I'll look into it when I got the access.

File: src/it/scala/org/ergoplatform/it/Docker.scala [hunk 1]
```diff
@@ -77,7 +77,11 @@ class Docker(suiteConfig: Config = ConfigFactory.empty,
     val networkPort = settings.scorexSettings.network.bindAddress.getPort
 
     val containerConfig: ContainerConfig = buildContainerConfig(nodeConfig, ip, restApiPort, networkPort)
-    val containerId = client.createContainer(containerConfig, configuredNodeName + "-" + hashString).id()
+    val containerId = Try {
+      client.createContainer(containerConfig, configuredNodeName + "-" + hashString)
+    }.recoverWith { case e: ImageNotFoundException =>
+      Failure(new Exception(s"Error: docker image is missing ($e)\nRun 'sbt it:test' to generate it."))
```
Comment (mike-aksarin) [inline src/it/scala/org/ergoplatform/it/Docker.scala]: maybe it worth nesting ImageNotFoundException to the newly created one?

Comment (greenhat) [reply src/it/scala/org/ergoplatform/it/Docker.scala]: It does not look nice, but we blow up in the next line anyway. It supposed to be a "help" message to just give a hint on how to fix it. Original exception is shown in log via string interpolation. However, I forgot to pass through other exception types. Fix is on the way.

File: src/it/scala/org/ergoplatform/it/Docker.scala [hunk 2]
```diff
@@ -79,8 +79,10 @@ class Docker(suiteConfig: Config = ConfigFactory.empty,
     val containerConfig: ContainerConfig = buildContainerConfig(nodeConfig, ip, restApiPort, networkPort)
     val containerId = Try {
       client.createContainer(containerConfig, configuredNodeName + "-" + hashString)
-    }.recoverWith { case e: ImageNotFoundException =>
-      Failure(new Exception(s"Error: docker image is missing ($e)\nRun 'sbt it:test' to generate it."))
+    }.recoverWith {
+      case e: ImageNotFoundException =>
+        Failure(new Exception(s"Error: docker image is missing ($e)\nRun 'sbt it:test' to generate it."))
```
Comment (mike-aksarin) [inline src/it/scala/org/ergoplatform/it/Docker.scala]: why don't you just pass it as a second parameter to the new Exception?

File: src/it/scala/org/ergoplatform/it/Docker.scala [hunk 3]
```diff
@@ -79,8 +79,10 @@ class Docker(suiteConfig: Config = ConfigFactory.empty,
...
+      case e => Failure(e)
```
Comment (mike-aksarin) [inline src/it/scala/org/ergoplatform/it/Docker.scala]: I don't think that you need it here. It is done by default if partial function doesn't match

File: src/it/scala/org/ergoplatform/it/Docker.scala (hunk 2 above)
Comment (greenhat) [reply src/it/scala/org/ergoplatform/it/Docker.scala]: Oh, sure. Sorry, I missed it the first time. Fixed.

File: src/it/scala/org/ergoplatform/it/Docker.scala (hunk 3 above)
Comment (greenhat) [reply src/it/scala/org/ergoplatform/it/Docker.scala]: Indeed. That compiler implementation of pf's `isDefinedAt` did not get under my skin deep enough yet. Thanks. Fixes.

**Pull request merged**